R/dataProcessPlots.R

```diff
@@ -129,20 +135,91 @@ dataProcessPlots = function(
+            you want to generate PDF using ggplot2")
+  }
+  
+  if (type == "PROFILEPLOT") {
+      plots <- .plotProfile(processed, summarized, featureName, ylimUp, ylimDown,
+                           x.axis.size, y.axis.size, text.size, text.angle, legend.size, 
+                           dot.size.profile, width, height, which.Protein, originalPlot, 
+                           summaryPlot, remove_uninformative_feature_outlier, address, isPlotly)
+      plotly_plots = list()
+      if(isPlotly) {
+          og_plotly_plot = NULL
+          summ_plotly_plot = NULL
+          if("original_plot" %in% names(plots)) {
+              for(i in seq_along(plots[["original_plot"]])) {
+                  plot_i <- plots[["original_plot"]][[paste("plot",i)]]
+                  og_plotly_plot <- .convertGgplot2Plotly(plot_i,tips=c("FEATURE","RUN","newABUNDANCE"))
+                  og_plotly_plot = .fixLegendPlotlyPlotsDataprocess(og_plotly_plot)
+                  og_plotly_plot = .fixCensoredPointsLegendProfilePlotsPlotly(og_plotly_plot)
+
+                  if(toupper(featureName) == "NA") {
+                      og_plotly_plot = .retainCensoredDataPoints(og_plotly_plot)
+                  }
+                  plotly_plots = c(plotly_plots, list(og_plotly_plot))
+              }
+          }
+          if("summary_plot" %in% names(plots)) {
+              for(i in seq_along(plots[["summary_plot"]])) {
+                  plot_i <- plots[["summary_plot"]][[paste("plot",i)]]
+                  summ_plotly_plot <- .convertGgplot2Plotly(plot_i,tips=c("FEATURE","RUN","ABUNDANCE"))
+                  summ_plotly_plot = .fixLegendPlotlyPlotsDataprocess(summ_plotly_plot)
+                  summ_plotly_plot = .fixCensoredPointsLegendProfilePlotsPlotly(summ_plotly_plot)
+                  if(toupper(featureName) == "NA") {
+                      summ_plotly_plot = .retainCensoredDataPoints(summ_plotly_plot)
+                  }
+                  plotly_plots = c(plotly_plots, list(summ_plotly_plot))
+              }
+          }
+          
+          if(address != FALSE) {
+              .savePlotlyPlotHTML(plotly_plots,address,"ProfilePlot" ,width, height)
+          }
+          plotly_plots
+      }
+  }
+    
+  else if (type == "QCPLOT") {
+      plots <- .plotQC(processed, featureName, ylimUp, ylimDown, x.axis.size, y.axis.size, 
+                      text.size, text.angle, legend.size, dot.size.profile, width, height,
+                      which.Protein, address, isPlotly)
+      plotly_plots <- vector("list", length(plots))
+      if(isPlotly) {
+          for(i in seq_along(plots)) {
+              plot <- plots[[i]]
+              plotly_plot <- .convertGgplot2Plotly(plot)
+              plotly_plot = .fixLegendPlotlyPlotsDataprocess(plotly_plot)
+              plotly_plots[[i]] = list(plotly_plot)
+          }
+            if(address != FALSE) {
+                .savePlotlyPlotHTML(plotly_plots,address,"QCPlot" ,width, height)
+            }
```

Comment:
This code snippet seems to be duplicated in multiple places.  Could we generalize the code into a helper method?

Reply:
let's generalize